roaring/containers_btree.go

```diff
@@ -142,6 +142,13 @@ func (btc *bTreeContainers) Clone() Containers {
 	return nbtc
 }
 
+func (btc *bTreeContainers) First() (key uint64, c *Container) {
+	if btc.tree.Len() == 0 {
+		return 0, nil
+	}
+	return btc.tree.First()
```

Comment:
I think this will subtly be broken in some contexts by the rowcache code, because I think there are some circumstances where the rowcache code lets a nil container sneak into the btree for a while. I don't think that's necessarily a bug in this code; probably it means that the rowcache code has to be stricter about ensuring that nil containers don't end up in the tree. (It turns out that the thing where an in-place operation can result in a nil container somewhat breaks iterators, though, if we try to delete the nil containers instead of allowing them in temporarily.)

Reply:
As far as I see most `bTreeContainers` methods assume `btc.tree` is not `nil`. `First` method has almost the same code with `Last`. Doesn't `Last` cause any problems in the rowcache code?

Reply:
I'm not sure. We should possibly file a separate issue to check through that. And actually it might have gotten fixed, it might be the SliceContainers that wasn't dealing with it well. My memory is fuzzy.